core/hmap.go

```diff
@@ -0,0 +1,39 @@
+package core
+
+import (
+	"errors"
+
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type HMap = *orderedmap.OrderedMap[string, string]
```

Comment:
Any specific reasons for using `orderedMap`? Asking as Redis doesn't [preserve order](https://stackoverflow.com/questions/40643482/redis-inserting-out-of-order-or-sorting-oddly) in case of `HSET`